app/src/main/java/org/wikipedia/talk/TalkTopicHolder.kt

```diff
@@ -23,18 +24,19 @@ class TalkTopicHolder internal constructor(
 ) : RecyclerView.ViewHolder(binding.root), View.OnClickListener {
 
     private val unreadTypeface = Typeface.create("sans-serif-medium", Typeface.NORMAL)
-    private var id: Int = 0
-    private var indicatorSha: String = ""
+    private var id: String = ""
 
-    fun bindItem(topic: TalkPage.Topic, searchQuery: String? = null) {
-        id = topic.id
-        indicatorSha = topic.getIndicatorSha()
-        val seen = AppDatabase.instance.talkPageSeenDao().getTalkPageSeen(topic.getIndicatorSha()) != null
-        var titleStr = RichTextUtil.stripHtml(topic.html).trim()
+    fun bindItem(threadItem: ThreadItem, searchQuery: String? = null) {
+        id = threadItem.id
+//        val seen = AppDatabase.instance.talkPageSeenDao().getTalkPageSeen(topic.getIndicatorSha()) != null
+        // TODO: implement seen
+        val seen = false
+
+        var titleStr = RichTextUtil.stripHtml(threadItem.html).trim()
         if (titleStr.isEmpty()) {
             // build up a title based on the contents, massaging the html into plain text that
             // flows over a few lines...
-            topic.replies?.firstOrNull()?.let {
+            threadItem.replies.firstOrNull()?.let {
                 titleStr = RichTextUtil.stripHtml(it.html).replace("\n", " ")
                 if (titleStr.length > MAX_CHARS_NO_SUBJECT) {
                     titleStr = titleStr.substring(0, MAX_CHARS_NO_SUBJECT) + "…"
```

Comment:
Would it be simpler to let the `TextView` handle ellipsizing the text on its own, instead of doing it manually ourselves?